File: ReadyTasks/CSharp/ProjectEuler/Task_35/Task_35/Program.cs
using System;

namespace Task_35
{
    class Program
    {
        static bool IsPrime(int numb)
        {
            for (int i = 2; i * i <= numb; i++)
            {
                if (numb % i == 0)
                {
                    return false;
                }
            }
            return numb > 1;
        }

        static int GetDigitNumb(int numb)
        {
            int result = 0;
            while (numb > 0)
            {
                result++;
                numb /= 10;
            }
            return result;
        }

        static bool IsCircularPrime(int numb)
        {
            int mult = 1;
            int digitCount = GetDigitNumb(numb);
            int power = (int)Math.Pow(10, digitCount - 1);

            while (digitCount > 0)
            {
                if (!IsPrime(numb))
                {
                    return false;
                }

                numb = numb / 10 + (numb % 10 * power);
                digitCount--;
            }

            return true;
        }

        static int GetCircularPrimesNumb(int max)
        {
            int result = 0;
            for (int i = 1; i <= max; i++)
            {
                if (IsCircularPrime(i))
                {
                    result++;
                }
            }
            return result;
        }

        static void Main(string[] args)
        {
            Console.WriteLine(GetCircularPrimesNumb(1000000));
        }
    }
}

File: ReadyTasks/FSharp/ProjectEuler/Task_35/Task_35/Program.fs
open System

let isPrime num =
    if (num < 2) then
        false
    else
        Seq.initInfinite (fun x -> x + 2) |> 
        Seq.takeWhile (fun x -> x * x <= num) |> 
        Seq.forall (fun x -> num % x > 0)

let isCircularPrime num =
    let generateNums numb =   
        let power = numb.ToString() |> String.length
        let mult = pown 10 (power - 1)
        [1..(power - 1)] |> List.scan (fun acc x -> acc / 10 + (acc % 10) * mult) numb
    generateNums num |> List.forall isPrime

let getResult numb =
    [1..numb] |> List.filter isCircularPrime |> List.length

[<EntryPoint>]
let main argv =
    printfn "%A" (getResult 1000000)
    0 // return an integer exit code